src/containers/Topics.js

```diff
@@ -0,0 +1,159 @@
+        />
+      )
+    }
+    return _.map(posts, _buildPostJSX)
+  }
+
+  _buildTopicBoxes(topics) {
+    const _buildTopicBox = (item, index='') => (
+      <TopicItem
+        key={`topic-item-${_.get(item, 'id', index)}`}
+        title={_.get(item, 'title', '')}
+        updatedAt={date2yyyymmdd(_.get(item, 'updated_at'), '.') || ''}
+        description={_.get(item, 'og_description', '')}
+        imgUrl={_.get(item, 'og_image.resized_targets.mobile.url', '') || _.get(item, 'leading_image.resized_targets.mobile.url', '') }
+        imgAlt={_.get(item, 'og_image.description', '') || _.get(item, 'leading_image.description', '')}
+        isFull={index === 0}
+        url={`${LINK_PREFIX.TOPICS}${_.get(item, 'slug')}`}
+      />
+    )
+    return _.map(topics, _buildTopicBox)
+  }
+
+  render() {
+    const { items, total, topicEntities, postEntities, topicListError, topicError } = this.props
+    const topics = denormalizeTopics(items, topicEntities, postEntities)
+    const topicsLength = topics.length
+    if (topicsLength <= 0) {
+      if (topicListError) {
+        return (<div><SystemError error={topicListError} /></div>)
+      }
+      return null
+    }
+    const topicsJSX = this._buildTopicBoxes(topics)
+    const topTopicJSX = topicsJSX[0]
+    const otherTopicsJSX = !topicError ? topicsJSX.slice(1) : topTopicJSX
+    const topRelatedPosts = _.get(topics, [ 0, 'relateds' ], []).slice(0, 3)
+    const topTopicName = _.get(topics, [ 0, 'topic_name' ], '')
+    const topTopicSlug = _.get(topics, [ 0, 'slug' ], '')
+    const hasMore = topicsLength < total
+    return (
+      <PageContainer>
+        <PageContent>
+          {topicError ? null : (
+            <TopSection topicName={topTopicName} topicUrl={`${LINK_PREFIX.TOPICS}${topTopicSlug}`}>
+              {topTopicJSX}
+              <PostsContainer>
+                {this._buildRelatedPosts(topRelatedPosts)}
+              </PostsContainer>
+            </TopSection>)}
+          <ListSection>
+            {otherTopicsJSX}
+          </ListSection>
+        </PageContent>
+        {!hasMore ? null : (
+          <div>
+            <More loadMore={this._loadMore}>
+              <span style={{ color: topicListError ? 'red' : 'white' }}>{topicListError ? '載入更多（請重試）' : '載入更多'}</span>
+            </More>
+          </div>
+        )}
+      </PageContainer>
+    )
+  }
+}
+
+function mapStateToProps(state) {
+  const items = _.get(state, [ reduxStateFields.topicList, 'items' ], [])
+  const total = _.get(state, [ reduxStateFields.topicList, 'total' ])
+  const topicListError = _.get(state, [ reduxStateFields.topicList, 'error' ], null)
+  const topicError = _.get(state, [ reduxStateFields.selectedTopic, 'error' ], null)
+  const entities = _.get(state, reduxStateFields.entities, {})
+  const topicEntities = _.get(entities, reduxStateFields.topicsInEntities, {})
+  const postEntities = _.get(entities, reduxStateFields.postsInEntities, {})
```

Comment:
If you could, compose/format the topic items you need here.
And just pass the `items` into `Topics`.
Decoupling the rendering logic and data composing logic will make `Topics` more flexible.